ls1intum/Artemis
# `Development`: Refactor exercise import components

### Checklist
#### General
#### Client

### Motivation and Context
The different import components in the client were all just copy-and-paste components.

### Description
This PR refactors the import components to use a common abstract superclass.

This PR focusses on the exercise imports, other components like `LectureImportComponent` will be part of a followup.

I directly included a small bugfix: It was not possible to sort by exam title, trying this would lead to an empty search result.

### Steps for Testing
Check that all exercise import components still work:
Prerequisites:
- 1 Instructor
- 1 Course with exercises of each type

1. Navigate to Course Management  -> Course -> Exercises
2. Import exercises of each type and make sure the import still works.
3. Play around with the different sorting options and the course / exam filter and make sure that everything works.

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Exam

1. Navigate to the exam management page
2. Check that importing exercises into the different exercise groups of the exam still works (for all exercise types)
3. Play around with the different sorting options and the course / exam filter and make sure that everything works.

### Review Progress

#### Code Review
#### Manual Tests
#### Exam Mode Test
